static/js/ace-editor.ts

```diff
@@ -1,230 +1,185 @@
-    }
-  });
-  window.addEventListener ('keyup', function (ev) {
-    const keyCode = ev.keyCode;
-    if (keyCode === 18) {
-      altPressed = false;
-      return;
-    }
-  });
-    return editor;
-  }
-
-  initializeExampleEditor(preview: HTMLElement): HedyEditor {
-    const dir = $("body").attr("dir");
-    const exampleEditor = this.turnIntoEditor(preview, true);
-    // Fits to content size
-    exampleEditor.setOptions({ maxLines: Infinity });
-    if ($(preview).hasClass('common-mistakes')) {
-      exampleEditor.setOptions({
-        showGutter: true,
-        showPrintMargin: true,
-        highlightActiveLine: true,
-        minLines: 5,
-      });
-    } else if ($(preview).hasClass('cheatsheet')) {
-      exampleEditor.setOptions({ minLines: 1 });
-    } else if ($(preview).hasClass('parsons')) {
-      exampleEditor.setOptions({
-        minLines: 1,
-        showGutter: false,
-        showPrintMargin: false,
-        highlightActiveLine: false
-      });
-    } else {
-      exampleEditor.setOptions({ minLines: 2 });
+      this.setHighlighterForLevel(theLevel)
     }
 
     if (dir === "rtl") {
-        exampleEditor.setOptions({ rtl: true });
+      this._editor.setOptions({ rtl: true });
     }
-    return exampleEditor;
   }
-}
-
-export class HedyAceEditor implements HedyEditor {
-  private _editor?: AceAjax.Editor;
-  private _markers?: Markers
-  askPromptOpen: boolean = false;
-
+  
   /**
- * Set the highlither rules for a particular level
- * @param level      
- */
-  setHighliterForLevel(level: number): void { 
+  * Set the highlither rules for a particular level
+  * @param level      
+  */
+  setHighlighterForLevel(level: number): void { 
     const mode = this.getHighlighter(level);
-    this._editor?.session.setMode(mode);
+    this._editor.session.setMode(mode);
   }
 
   /**
    * @returns the string of the current program in the editor
    */
-  getValue(): string { 
-    return this._editor!.getValue();
-  }
-
-  /**     
-   * @returns if the editor is set to read-only mode
-   */
-  isReadOnly(): boolean {
-    return this._editor!.getReadOnly();
+  public get contents(): string { 
+    // Always trim trailing whitespaces before returning the contents
+    try {
+      // This module may or may not exist, so let's be extra careful here.
+      const whitespace = ace.require("ace/ext/whitespace");
+      whitespace.trimTrailingSpace(this._editor.session, true);
+    } catch (e) {
+      console.error(e);
+    }
+    return this._editor.getValue();
   }
 
   /**
    * Sets the editor contents.
    * @param content the content that wants to be set in the editor
    */
-  setValue(content: string): void {
-    this._editor?.setValue(content, MOVE_CURSOR_TO_END);
+  public set contents(content: string) {
+    this._editor.setValue(content, MOVE_CURSOR_TO_END);
   }
 
+  /**     
+   * @returns if the editor is set to read-only mode
+   */
+  public get getIsReadOnly(): boolean {
```

Comment:
Probably

```ts
public get isReadOnly(): boolean {
  return this._isReadOnly;
}
```

?

Reply:
Or honestly, can you read directly from the editor you wrapped?

Reply:
You are so right. I'm using the editor to handle that now.